[CIAPP] Add `SARIF` upload command (#777)
DataDog/datadog-ci: src/commands/sarif/README.md

# SARIF command

Upload your SARIF report files.

## Usage

#### Commands

##### `upload`

This command will upload your SARIF report to Datadog.

```bash
datadog-ci sarif upload [--service] [--max-concurrency] [--dry-run] [--no-verify] [--tags] <paths>
```

For example:

```bash
datadog-ci sarif upload --service my-service --tags key1:value1 --tags key2:value2 sarif-reports/go-reports sarif-reports/java-reports sarif-report/single-report.sarif
```

- The positional arguments are the directories or file paths in which the SARIF reports are located. If you pass a folder, the CLI will look for all `.sarif` files in it.
- `--service` (default: `DD_SERVICE` env var) should be set as the name of the service you're uploading SARIF reports for.

> - `--tags` is a array of key value pairs of the shape `key:value`. This will set global tags applied to all results.
>   - The resulting dictionary will be merged with whatever is in the `DD_TAGS` environment variable. If a `key` appears both in `--tags` and `DD_TAGS`, whatever value is in `DD_TAGS` will take precedence.

Reviewer: ```suggestion
- `--tags` is a array of key value pairs of the form `key:value`. This parameter sets global tags applied to all results. The upload process merges the tags passed on the command line with the tags in the `DD_TAGS` environment variable. If a key appears in both `--tags` and `DD_TAGS`, the value in `DD_TAGS` takes precedence.
```

> - `--env` (default: `DD_ENV` env var) is a string that represents the environment where you want your tests to appear in.

Reviewer: ```suggestion
- `--env` (default: `DD_ENV` env var) is a string that represents the environment in which you want your tests to appear.
```

- `--max-concurrency` (default: `20`): number of concurrent uploads to the API.

> - `--dry-run` (default: `false`): it will run the command without the final upload step. All other checks are performed.
> - `--no-verify` (default: `false`): it will run the command without performing reports validation on the CLI.

Reviewer: ```suggestion
- `--dry-run` (default: `false`): runs the command without the final upload step. All other checks are performed.
- `--no-verify` (default: `false`): runs the command without performing report validation on the CLI.
```

#### Environment variables

> Additionally you might configure the `sarif` command with environment variables:

Reviewer: ```suggestion
Additionally, you may configure the `sarif` command with environment variables:
```
Similarly to above, this list would also be easier to read if it were formatted as a table or description list.
